### PR TITLE
INFRA 24707: add option to disable pop-ups

### PR DESCRIPTION
**Pull request by TheCamNelson**

### CODE REVIEW

File: inventory/environ.py [hunk 1]
```diff
@@ -403,6 +404,16 @@ def getASan(vars_scope):
     if vars_scope["splunk"]["asan"]:
         vars_scope["ansible_environment"].update({"ASAN_OPTIONS": "detect_leaks=0"})
 
+def getDisablePopups(vars_scope):
+    """
+    Configure pop-up settings
+    """
+    popups_disabled = os.environ.get("SPLUNK_DISABLE_POPUPS", "")
```
Comment (alishamayor) [inline inventory/environ.py:411]: Can't you just add this as another line in `overrideEnvironmentVars` below?

File: roles/splunk_common/tasks/disable_popups.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+      url: "{{ cert_prefix }}://127.0.0.1:{{ splunk.svc_port }}/{{ item.key }}"
+      method: POST
+      user: "{{ splunk.admin_user }}"
+      password: "{{ splunk.password }}"
+      body: "{{ item.value }}"
+      validate_certs: false
+      status_code: 200,201,409
+    with_items:
+      - { key: "servicesNS/admin/user-prefs/data/user-prefs/general", value: "hideInstrumentationOptInModal=1&notification_python_3_impact=false&showWhatsNew=0&dismissedInstrumentationOptInVersion=4" }
+      - { key: "servicesNS/nobody/splunk_instrumentation/admin/telemetry/general", value: "showOptInModal=0&optInVersionAcknowledged=4" }
```
Comment (nwang92) [inline roles/splunk_common/tasks/disable_popups.yml:13]: I would be careful about hardcoding this `optInVersionAcknowledged=4` here? Seems like this is being compared with `optInVersion` and that may be an unbounded number, so a future release may render this change ineffective.

https://docs.splunk.com/Documentation/Splunk/8.0.5/Admin/Telemetryconf

File: roles/splunk_common/tasks/disable_popups.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    uri:
+      url: "{{ cert_prefix }}://127.0.0.1:{{ splunk.svc_port }}/{{ item.key }}"
+      method: POST
+      user: "{{ splunk.admin_user }}"
+      password: "{{ splunk.password }}"
+      body: "{{ item.value }}"
+      validate_certs: false
+      status_code: 200,201,409
+    with_items:
+      - { key: "servicesNS/admin/user-prefs/data/user-prefs/general", value: "hideInstrumentationOptInModal=1&notification_python_3_impact=false&showWhatsNew=0&dismissedInstrumentationOptInVersion=4" }
```
Comment (nwang92) [inline roles/splunk_common/tasks/disable_popups.yml:12]: Ditto as below for `dismissedInstrumentationOptInVersion`

File: inventory/environ.py (hunk 1 above)
Comment (nwang92) [reply inventory/environ.py:411]: I'm fine either way, but give me some small tests please :)

**Comment by nwang92** [comment]
This feature flag seems like a good candidate to add to the molecule functional tests. Maybe add the new parameter in https://github.com/splunk/splunk-ansible/tree/develop/roles/splunk_standalone/molecule/snowflake and verify that the conf files are being generated as intended?

**Pull request merged**